src/pyedb/dotnet/edb_core/dotnet/primitive.py

```diff
@@ -376,6 +376,39 @@ def expand(self, offset=0.001, tolerance=1e-12, round_corners=True, maximum_corn
         self.polygon_data = new_poly[0]
         return True
 
+    def scale(self, factor, center=None):
+        """Scales the polygon relative to a center point by a factor.
+
+        Parameters
+        ----------
+        factor : float
+            Scaling factor.
+        center : List of float or str [x,y], optional
+            If None scaling is done from polygon center.
+
+        Returns
+        -------
+        bool
+           ``True`` when successful, ``False`` when failed.
+        """
+        if not isinstance(factor, str):
+            factor = float(factor)
+            polygon_data = self.polygon_data.create_from_arcs(self.polygon_data.edb_api.GetArcData(), True)
+            if not center:
+                center = self.polygon_data.edb_api.GetBoundingCircleCenter()
+                if center:
```

Comment:
Can you log an error when this if statement is not valid ? Or assume that it should work (I mean pass the if statement) and remove this condition ?

Reply:
Done